protocol/statetracker/state_query.go

```diff
@@ -68,6 +74,42 @@ func (csq *StateQuery) GetSpec(ctx context.Context, chainID string) (*spectypes.
 	return &spec.Spec, nil
 }
 
+func (csq *StateQuery) CheckEmergencyMode(ctx context.Context) (isEmergency bool, virtualEpoch uint64, err error) {
+	downtimeParams, err := csq.DowntimeClient.QueryParams(ctx, &downtimev1.QueryParamsRequest{})
```

Comment:
this method costs us 2 extra queries per block, which is increasing the amount of queries we are sending to the blockchain by x3 as we are currently sending only 1 per block. we can improve it by saving the downtime params on bootstrap, and creating an updater to update if there are param change events. you can take a look at version updater and spec updater are doing the same methodology. 

for the latestBlockTime, this method is being called upon block updates, so we can just say its time.now(). even if its not 100% accurate downtime wont trigger in matter of seconds so as long as we dont underestimate its ok (rather over estimate)